tests/test_data.py

```diff
@@ -63,6 +63,17 @@ def test_advanced_indexing(self):
         assert np.all(u.data[1, :, :, 0] == 0.)
         assert np.all(u.data[1, :, :, -1] == 0.)
 
+    @skipif('yask')
+    def test_negative_step(self):
+        """Test slicing with a negative step."""
+        grid = Grid(shape=(6, 6, 6))
+        u = TimeFunction(name='u', grid=grid, dtype=np.int32)
+        u.data[:] = 0.
+        dat = np.array([1, 2, 3, 4, 5, 6])
+        u.data[0, :, 0, 0] = dat
+        assert (np.array(u.data[0, 3::-1, 0, 0]) == dat[3::-1]).all()
```

Comment:
Wouldn't the test be more "improved" as 
`u.data[0, 3::-1, 0, 0] = dat[3::-1]`
`assert (np.array(u.data[0, :, 0, 0])  == dat[:]).all()`

Reply:
The 'full' length of dim 1 of `u.data` is not being set in your example and hence there's no reason for
`assert (np.array(u.data[0, :, 0, 0]) == dat[:]).all()`
to be true.

The test is meant to test cases when a negative slice is not starting from the end point so I think is fine as is?